docs/librarymanager.md

```diff
@@ -0,0 +1,491 @@
+
+```css
+.popup {
+
+}
+
+.popup > .popup-pane > .popup-region {
+    /* https://commons.wikimedia.org/wiki/File:Speech_bubble.svg */
+    -fx-shape: "M 45.673,0 C 67.781,0 85.703,12.475 85.703,27.862 C 85.703,43.249 67.781,55.724 45.673,55.724 C 38.742,55.724 32.224,54.497 26.539,52.34 C 15.319,56.564 0,64.542 0,64.542 C 0,64.542 9.989,58.887 14.107,52.021 C 15.159,50.266 15.775,48.426 16.128,46.659 C 9.618,41.704 5.643,35.106 5.643,27.862 C 5.643,12.475 23.565,0 45.673,0 M 45.673,2.22 C 24.824,2.22 7.862,13.723 7.862,27.863 C 7.862,34.129 11.275,40.177 17.472,44.893 L 18.576,45.734 L 18.305,47.094 C 17.86,49.324 17.088,51.366 16.011,53.163 C 15.67,53.73 15.294,54.29 14.891,54.837 C 18.516,53.191 22.312,51.561 25.757,50.264 L 26.542,49.968 L 27.327,50.266 C 32.911,52.385 39.255,53.505 45.673,53.505 C 66.522,53.505 83.484,42.002 83.484,27.862 C 83.484,13.722 66.522,2.22 45.673,2.22 L 45.673,2.22 z ";
+    -fx-background-color: red;
+}
+
+```
+</details>
+
+As it can be seen in the source code, the control is a regular JavaFX control with some JavaFX properties and a custom skin.
+
+There are some specific optional annotations intended for FXML:
+
+- `@DefaultProperty`: Specifies a property to which child elements will be added or set when an explicit property is not given.
+- `@NamedArg`: allows an FXMLLoader to instantiate a class that does not have a zero-argument constructor.
+
+When the control is added to Scene Builder, the control’s properties are visible in the Inspector area (to the right), and the user can interact with them:
+
+<p align="center"><img src="images/librarymanager/SceneBuilder20.png" alt="The popup custom control in action" width="400"></p>
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+
+<?import javafx.scene.layout.AnchorPane?>
+<?import javafx.scene.text.Font?>
+<?import popup.Popup?>
+
+<AnchorPane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="400.0" prefWidth="600.0" xmlns="http://javafx.com/javafx/21.0.1" xmlns:fx="http://javafx.com/fxml/1">
+   <children>
+      <Popup bubbleFill="#19ee2b" content="This is a message!" layoutX="351.0" layoutY="4.0" prefHeight="219.0" prefWidth="211.0" textFill="#dd0f0f">
+         <textFont>
+            <Font name="Bradley Hand Bold" size="25.0" />
+         </textFont>
+      </Popup>
+   </children>
+</AnchorPane>
+```
+
+### Troubleshooting
+
+After importing a jar with one or more possible custom controls, it is possible that not all the expected controls show up. 
+
+The Jar Analysis Report can provide insightful information about the reasons for this. As mentioned earlier, when exploring the jar, the FXMLLoader is instantiated with the possible custom control classes, and if something fails, the result is one or more exceptions that are captured by the Jar Analysis and the report just shows them:
+
+<p align="center"><img src="images/librarymanager/SceneBuilder21.png" alt="The Jar Analysis report" width="400"></p>
```

Comment:
As of now, all image are nicely readable. This one appears a little too small. Hence I would also change its size to 500px or 600px. (`SceneBuilder21.png`).

Reply:
done